api/v1beta1/vcdmachine_types.go

```diff
@@ -76,13 +76,13 @@ type VCDMachineSpec struct {
 	// ExtraOvdcNetworks is the list of extra Ovdc Networks that are mounted to machines.
 	// VCDClusterSpec.OvdcNetwork is always attached regardless of this field.
 	// +optional
-	ExtraOvdcNetworks []string `json:"extraOvdcNetworks"`
+	ExtraOvdcNetworks []string `json:"extraOvdcNetworks,omitempty"`
 
 	// VmNamingTemplate is go template to generate VM names based on Machine and VCDMachine CRs.
 	// Functions of Sprig library are supported. See https://github.com/Masterminds/sprig
 	// Immutable field. machine.Name is used as VM name when this field is empty.
 	// +optional
-	VmNamingTemplate string `json:"vmNamingTemplate"`
+	VmNamingTemplate string `json:"vmNamingTemplate,omitempty"`
```

Comment:
Users should be able to create VCDMachine CRs without providing empty values for these fields. Missing omitempty tags lead to errors in CAPI controllers while creating VCDMachine by using VCDMachineTemplate.